### PR TITLE
rewrite command of telescope

### PR DESCRIPTION
**Pull request by glepnir**
@tjdevries  now the usage like `Telescope find_files`. `Telescope grep_string`, If want change some options, like this 

`Telescope find_files prompt_prefix=ssss` .

current can't get theme option in viml, so does not support add theme option in command .

### CODE REVIEW

**Comment by tjdevries** [comment]
Can we add a completion function for the first item at least? So that it can complete to the correct builtin?

**Comment by tjdevries** [comment]
Also, do we want to leave the commands? I'm not sure if anyone is using them or not currently :/

**Comment by glepnir** [comment]
we need command. because it's useful for some layload feature like dein.  use `on_cmd` keyword  of dein to load telescope.just load telescope when call the command.

**Comment by tjdevries** [comment]
Sorry, I meant, do you want to leave the old commands you made? Or should we delete them.

**Comment by glepnir** [comment]
yep we should delete. they are hardcored and sees ugly.. done.

**Comment by OkanEsen** [comment]
This'll be a breaking change though and "seems" ugly is kind of subjective, no? :)

I think this change should be documented on the README at least or should probably throw an error, that the old commands are deprecated in favor of the new syntax?

**Comment by glepnir** [comment]
Sorry for the impact on you, I didn’t consider more settings when I made them, so I rewrote them to make them more friendly.

If you use some lazyload plugin management like dein just config `on_cmd = 'Telescope'`,  And If you want  call them in commandline just ` Telescope ` then press `tab` .and support custom options of telescope

**Comment by OkanEsen** [comment]
Don't be sorry, it's all good and doesn't affect me, since I use my own mappings likes this: 

``` lua
utils.gmap(
  "n",
  "xxx",
  [[<cmd>lua require'plugin.telescope'.foo_bar()<CR>]],
  {...}
)
```

I was just making a comment for the people this change could affect and that's why I suggested, that this change should probably be documented on the README. :)

**Comment by glepnir** [comment]
The command is only effective for most people who use lazy loading. If load telescope normally, Just need to map `lua require`.

**Comment by tami5** [comment]
This is very useful change, I like it more then the previous way. Thanks @glepnir

**Comment by glepnir** [comment]
@tjdevries cc/

**Comment by tjdevries** [comment]
@glepnir can you add documentation to readme?

Also, for completion, you can use luaeval and vim.tbl_keys to get the keys, rather than hard coding. 

Then I will merge.

**Comment by glepnir** [comment]
@tjdevries  Done.

File: plugin/telescope.vim [hunk 1]
```diff
@@ -28,25 +28,30 @@ cnoremap <silent> <Plug>(TelescopeFuzzyCommandSearch) <C-\>e
...
-command! -nargs=0 -bar TelescopeLspCodeActions lua require'telescope.builtin'.lsp_code_actions{}
-command! -nargs=0 -bar TelescopeTreesitter lua require'telescope.builtin'.treesitter{}
-command! -nargs=0 -bar TelescopePlanets lua require'telescope.builtin'.planets{}
-command! -nargs=0 -bar TelescopeHelpTags lua require'telescope.builtin'.help_tags{}
-command! -nargs=0 -bar TelescopeManPages lua require'telescope.builtin'.man_pages{}
-command! -nargs=0 -bar TelescopeColorscheme lua require'telescope.builtin'.colorscheme{}
-command! -nargs=0 -bar TelescopeMarks lua require'telescope.builtin'.marks{}
+" Telescope builtin lists
+function! s:telescope_complete(...)
+  let l:telescope_builtin = luaeval('require("telescope.builtin")')
```
Comment (tjdevries) [inline plugin/telescope.vim:33]: In luaeval, you should just return the list of keys. I don't know that this table will always serialize.

Use vim.tbl_keys function.

**Review by tjdevries — CHANGES_REQUESTED** [review]
Small change, then it's good.

**Comment by glepnir** [comment]
Done. what about this?

**Comment by tjdevries** [comment]
Sorry, I meant within the Lua eval call, inside the string.

I'm on mobile so it's too hard to type it at the moment 😂

**Comment by glepnir** [comment]
@tjdevries  haha It's Ok. I know your means. Done.

**Comment by tjdevries** [comment]
Thanks for fixing those things up! As always, thanks for the good ideas 😁

**Pull request merged**